src/lib/session.ts

```diff
@@ -1,40 +1,30 @@
 import { derived, get, writable } from "svelte/store";
 
-export interface Session {
-  id: string;
-  publicKey: string;
-}
+import { HttpdClient } from "@httpd-client";
 
-interface SessionResponse {
-  sessionId: string;
-  status: string;
+export interface StoredSession {
+  id: string;
   publicKey: string;
-  issuedAt: number;
-  expiresAt: number;
 }
 
-const store = writable<Session | undefined>(undefined);
+const store = writable<StoredSession | undefined>(undefined);
 export const sessionStore = derived(store, s => s);
 
-const endpoint = "http://localhost:8080/api/v1/sessions";
+const api = new HttpdClient("http://localhost:8080");
 
 export async function authenticate(params: {
   id: string;
   signature: string;
   publicKey: string;
-}): Promise<"success" | "failure"> {
+}): Promise<boolean> {
   disconnect();
```

Comment:
There’s an `await` missing here which may lead to race conditions.

Reply:
Good catch! abd623bdf15aab941d042c59e2c3331f9bb8a6a0

Was there a way to catch missing `await`s with eslint or something like that?

Reply:
There is a [no-floating-promises rule](https://typescript-eslint.io/rules/no-floating-promises/) that catches these issues. But you do need to enable type support for Eslint for this to work

Reply:
https://github.com/radicle-dev/radicle-interface/pull/674